sdk/monitor/Azure.Monitor.Query/src/MetricsQueryResourcesOptions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.Monitor.Query.Models;
+
+namespace Azure.Monitor.Query
+{
+    /// <summary>
+    /// Provides the options for the MetricsClient QueryResource method.
+    /// </summary>
+    public class MetricsQueryResourcesOptions
+    {
+        /// <summary>
+        /// Gets or sets the timespan over which the metric will be queried.
+        /// </summary>
+        [CodeGenMember("TimeSpan")]
+        public QueryTimeRange? TimeRange { get; set; }
+
+        /// <summary>
+        /// <para>
+        /// Gets the list of metric aggregations to retrieve.
+        /// </para>
+        /// <para>
+        /// Although this collection cannot be set, it can be modified.
+        /// See <see href="https://learn.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/object-and-collection-initializers#object-initializers-with-collection-read-only-property-initialization">Object initializers with collection read-only property initialization</see>.
+        /// </para>
+        /// </summary>
+        public IEnumerable<MetricAggregationType> Aggregations { get; set; }
```

Comment:
I think collection properties should not have setters, but it is also doesn't seem like users would be able to modify this collection without casting because of how we are defining the type.

Reply:
should I make it internal set?